// @porsche-design-system/components-angular/fesm2022/porsche-design-system-components-angular.mjs
import { load } from '@porsche-design-system/components-js';
export { componentsReady } from '@porsche-design-system/components-js';
import * as i0 from '@angular/core';
import { InjectionToken, Component, ChangeDetectionStrategy, inject, EventEmitter, NgModule, Optional, Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';

const THEME_TOKEN = new InjectionToken('pdsTheme');
class BaseComponent {
    el;
    constructor(cdr, elementRef) {
        cdr.detach();
        this.el = elementRef.nativeElement;
    }
    ngOnChanges(props) {
        for (const prop in props) {
            this.el[prop] = props[prop].currentValue;
        }
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: BaseComponent, deps: [{ token: i0.ChangeDetectorRef }, { token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: BaseComponent, isStandalone: true, selector: "ng-component", usesOnChanges: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: BaseComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: () => [{ type: i0.ChangeDetectorRef }, { type: i0.ElementRef }] });
class BaseComponentWithTheme extends BaseComponent {
    theme;
    themeSubscription = inject(THEME_TOKEN).subscribe((theme) => {
        this.el.theme = this.theme || theme;
    });
    ngOnDestroy() {
        // need to manually unsubscribe or otherwise subscription is still active even after unmount
        // https://rafaelneto.dev/en/blog/unsubscribing-behaviorsubject-observable-angular/
        this.themeSubscription.unsubscribe();
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: BaseComponentWithTheme, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: BaseComponentWithTheme, isStandalone: true, selector: "ng-component", usesInheritance: true, ngImport: i0, template: '', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: BaseComponentWithTheme, decorators: [{
            type: Component,
            args: [{
                    template: '',
                }]
        }] });

class PAccordion extends BaseComponentWithTheme {
    compact;
    heading;
    headingTag;
    open;
    size;
    sticky;
    /** @deprecated */
    tag;
    theme;
    /** @deprecated */
    accordionChange = new EventEmitter();
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PAccordion, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PAccordion, isStandalone: false, selector: "p-accordion,[p-accordion]", inputs: { compact: "compact", heading: "heading", headingTag: "headingTag", open: "open", size: "size", sticky: "sticky", tag: "tag", theme: "theme" }, outputs: { accordionChange: "accordionChange", update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PAccordion, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-accordion,[p-accordion]',
                    template: '<ng-content />',
                    inputs: ['compact', 'heading', 'headingTag', 'open', 'size', 'sticky', 'tag', 'theme'],
                    outputs: ['accordionChange', 'update'],
                    standalone: false
                }]
        }] });

class PBanner extends BaseComponentWithTheme {
    description;
    dismissButton;
    heading;
    headingTag;
    open;
    /** @deprecated */
    persistent;
    state;
    theme;
    /** @deprecated */
    width;
    dismiss = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PBanner, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PBanner, isStandalone: false, selector: "p-banner,[p-banner]", inputs: { description: "description", dismissButton: "dismissButton", heading: "heading", headingTag: "headingTag", open: "open", persistent: "persistent", state: "state", theme: "theme", width: "width" }, outputs: { dismiss: "dismiss" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PBanner, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-banner,[p-banner]',
                    template: '<ng-content />',
                    inputs: ['description', 'dismissButton', 'heading', 'headingTag', 'open', 'persistent', 'state', 'theme', 'width'],
                    outputs: ['dismiss'],
                    standalone: false
                }]
        }] });

class PButton extends BaseComponentWithTheme {
    aria;
    compact;
    disabled;
    form;
    hideLabel;
    icon;
    iconSource;
    loading;
    name;
    theme;
    type;
    value;
    variant;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PButton, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PButton, isStandalone: false, selector: "p-button,[p-button]", inputs: { aria: "aria", compact: "compact", disabled: "disabled", form: "form", hideLabel: "hideLabel", icon: "icon", iconSource: "iconSource", loading: "loading", name: "name", theme: "theme", type: "type", value: "value", variant: "variant" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PButton, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-button,[p-button]',
                    template: '<ng-content />',
                    inputs: ['aria', 'compact', 'disabled', 'form', 'hideLabel', 'icon', 'iconSource', 'loading', 'name', 'theme', 'type', 'value', 'variant'],
                    standalone: false
                }]
        }] });

class PButtonGroup extends BaseComponent {
    direction;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PButtonGroup, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PButtonGroup, isStandalone: false, selector: "p-button-group,[p-button-group]", inputs: { direction: "direction" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PButtonGroup, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-button-group,[p-button-group]',
                    template: '<ng-content />',
                    inputs: ['direction'],
                    standalone: false
                }]
        }] });

class PButtonPure extends BaseComponentWithTheme {
    active;
    alignLabel;
    aria;
    disabled;
    form;
    hideLabel;
    icon;
    iconSource;
    loading;
    name;
    size;
    stretch;
    theme;
    type;
    underline;
    value;
    /** @deprecated */
    weight;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PButtonPure, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PButtonPure, isStandalone: false, selector: "p-button-pure,[p-button-pure]", inputs: { active: "active", alignLabel: "alignLabel", aria: "aria", disabled: "disabled", form: "form", hideLabel: "hideLabel", icon: "icon", iconSource: "iconSource", loading: "loading", name: "name", size: "size", stretch: "stretch", theme: "theme", type: "type", underline: "underline", value: "value", weight: "weight" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PButtonPure, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-button-pure,[p-button-pure]',
                    template: '<ng-content />',
                    inputs: ['active', 'alignLabel', 'aria', 'disabled', 'form', 'hideLabel', 'icon', 'iconSource', 'loading', 'name', 'size', 'stretch', 'theme', 'type', 'underline', 'value', 'weight'],
                    standalone: false
                }]
        }] });

class PButtonTile extends BaseComponent {
    align;
    aria;
    aspectRatio;
    background;
    compact;
    description;
    disabled;
    gradient;
    icon;
    iconSource;
    label;
    loading;
    size;
    type;
    weight;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PButtonTile, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PButtonTile, isStandalone: false, selector: "p-button-tile,[p-button-tile]", inputs: { align: "align", aria: "aria", aspectRatio: "aspectRatio", background: "background", compact: "compact", description: "description", disabled: "disabled", gradient: "gradient", icon: "icon", iconSource: "iconSource", label: "label", loading: "loading", size: "size", type: "type", weight: "weight" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PButtonTile, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-button-tile,[p-button-tile]',
                    template: '<ng-content />',
                    inputs: ['align', 'aria', 'aspectRatio', 'background', 'compact', 'description', 'disabled', 'gradient', 'icon', 'iconSource', 'label', 'loading', 'size', 'type', 'weight'],
                    standalone: false
                }]
        }] });

class PCanvas extends BaseComponentWithTheme {
    sidebarEndOpen;
    sidebarStartOpen;
    theme;
    sidebarEndDismiss = new EventEmitter();
    sidebarStartUpdate = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PCanvas, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PCanvas, isStandalone: false, selector: "p-canvas,[p-canvas]", inputs: { sidebarEndOpen: "sidebarEndOpen", sidebarStartOpen: "sidebarStartOpen", theme: "theme" }, outputs: { sidebarEndDismiss: "sidebarEndDismiss", sidebarStartUpdate: "sidebarStartUpdate" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PCanvas, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-canvas,[p-canvas]',
                    template: '<ng-content />',
                    inputs: ['sidebarEndOpen', 'sidebarStartOpen', 'theme'],
                    outputs: ['sidebarEndDismiss', 'sidebarStartUpdate'],
                    standalone: false
                }]
        }] });

class PCarousel extends BaseComponentWithTheme {
    activeSlideIndex;
    alignControls;
    alignHeader;
    aria;
    description;
    /** @deprecated */
    disablePagination;
    focusOnCenterSlide;
    gradientColor;
    heading;
    headingSize;
    intl;
    pagination;
    rewind;
    skipLinkTarget;
    slidesPerPage;
    theme;
    trimSpace;
    width;
    /** @deprecated */
    wrapContent;
    /** @deprecated */
    carouselChange = new EventEmitter();
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PCarousel, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PCarousel, isStandalone: false, selector: "p-carousel,[p-carousel]", inputs: { activeSlideIndex: "activeSlideIndex", alignControls: "alignControls", alignHeader: "alignHeader", aria: "aria", description: "description", disablePagination: "disablePagination", focusOnCenterSlide: "focusOnCenterSlide", gradientColor: "gradientColor", heading: "heading", headingSize: "headingSize", intl: "intl", pagination: "pagination", rewind: "rewind", skipLinkTarget: "skipLinkTarget", slidesPerPage: "slidesPerPage", theme: "theme", trimSpace: "trimSpace", width: "width", wrapContent: "wrapContent" }, outputs: { carouselChange: "carouselChange", update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PCarousel, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-carousel,[p-carousel]',
                    template: '<ng-content />',
                    inputs: ['activeSlideIndex', 'alignControls', 'alignHeader', 'aria', 'description', 'disablePagination', 'focusOnCenterSlide', 'gradientColor', 'heading', 'headingSize', 'intl', 'pagination', 'rewind', 'skipLinkTarget', 'slidesPerPage', 'theme', 'trimSpace', 'width', 'wrapContent'],
                    outputs: ['carouselChange', 'update'],
                    standalone: false
                }]
        }] });

class PCheckbox extends BaseComponentWithTheme {
    checked;
    compact;
    disabled;
    form;
    hideLabel;
    indeterminate;
    label;
    loading;
    message;
    name;
    required;
    state;
    theme;
    value;
    blur = new EventEmitter();
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PCheckbox, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PCheckbox, isStandalone: false, selector: "p-checkbox,[p-checkbox]", inputs: { checked: "checked", compact: "compact", disabled: "disabled", form: "form", hideLabel: "hideLabel", indeterminate: "indeterminate", label: "label", loading: "loading", message: "message", name: "name", required: "required", state: "state", theme: "theme", value: "value" }, outputs: { blur: "blur", update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PCheckbox, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-checkbox,[p-checkbox]',
                    template: '<ng-content />',
                    inputs: ['checked', 'compact', 'disabled', 'form', 'hideLabel', 'indeterminate', 'label', 'loading', 'message', 'name', 'required', 'state', 'theme', 'value'],
                    outputs: ['blur', 'update'],
                    standalone: false
                }]
        }] });

class PCheckboxWrapper extends BaseComponentWithTheme {
    hideLabel;
    label;
    loading;
    message;
    state;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PCheckboxWrapper, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PCheckboxWrapper, isStandalone: false, selector: "p-checkbox-wrapper,[p-checkbox-wrapper]", inputs: { hideLabel: "hideLabel", label: "label", loading: "loading", message: "message", state: "state", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PCheckboxWrapper, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-checkbox-wrapper,[p-checkbox-wrapper]',
                    template: '<ng-content />',
                    inputs: ['hideLabel', 'label', 'loading', 'message', 'state', 'theme'],
                    standalone: false
                }]
        }] });

/** @deprecated since v3.0.0, will be removed with next major release. Use native CSS Grid instead. */
class PContentWrapper extends BaseComponentWithTheme {
    /** @deprecated */
    backgroundColor;
    /** @deprecated */
    theme;
    width;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PContentWrapper, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PContentWrapper, isStandalone: false, selector: "p-content-wrapper,[p-content-wrapper]", inputs: { backgroundColor: "backgroundColor", theme: "theme", width: "width" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PContentWrapper, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-content-wrapper,[p-content-wrapper]',
                    template: '<ng-content />',
                    inputs: ['backgroundColor', 'theme', 'width'],
                    standalone: false
                }]
        }] });

class PCrest extends BaseComponent {
    aria;
    href;
    target;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PCrest, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PCrest, isStandalone: false, selector: "p-crest,[p-crest]", inputs: { aria: "aria", href: "href", target: "target" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PCrest, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-crest,[p-crest]',
                    template: '<ng-content />',
                    inputs: ['aria', 'href', 'target'],
                    standalone: false
                }]
        }] });

class PDisplay extends BaseComponentWithTheme {
    align;
    color;
    ellipsis;
    size;
    tag;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PDisplay, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PDisplay, isStandalone: false, selector: "p-display,[p-display]", inputs: { align: "align", color: "color", ellipsis: "ellipsis", size: "size", tag: "tag", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PDisplay, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-display,[p-display]',
                    template: '<ng-content />',
                    inputs: ['align', 'color', 'ellipsis', 'size', 'tag', 'theme'],
                    standalone: false
                }]
        }] });

class PDivider extends BaseComponentWithTheme {
    color;
    direction;
    /** @deprecated */
    orientation;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PDivider, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PDivider, isStandalone: false, selector: "p-divider,[p-divider]", inputs: { color: "color", direction: "direction", orientation: "orientation", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PDivider, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-divider,[p-divider]',
                    template: '<ng-content />',
                    inputs: ['color', 'direction', 'orientation', 'theme'],
                    standalone: false
                }]
        }] });

class PFieldset extends BaseComponentWithTheme {
    label;
    labelSize;
    message;
    required;
    state;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFieldset, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PFieldset, isStandalone: false, selector: "p-fieldset,[p-fieldset]", inputs: { label: "label", labelSize: "labelSize", message: "message", required: "required", state: "state", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFieldset, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-fieldset,[p-fieldset]',
                    template: '<ng-content />',
                    inputs: ['label', 'labelSize', 'message', 'required', 'state', 'theme'],
                    standalone: false
                }]
        }] });

/** @deprecated since v3.0.0, will be removed with next major release. Please use "p-fieldset" instead. */
class PFieldsetWrapper extends BaseComponentWithTheme {
    label;
    labelSize;
    message;
    required;
    state;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFieldsetWrapper, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PFieldsetWrapper, isStandalone: false, selector: "p-fieldset-wrapper,[p-fieldset-wrapper]", inputs: { label: "label", labelSize: "labelSize", message: "message", required: "required", state: "state", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFieldsetWrapper, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-fieldset-wrapper,[p-fieldset-wrapper]',
                    template: '<ng-content />',
                    inputs: ['label', 'labelSize', 'message', 'required', 'state', 'theme'],
                    standalone: false
                }]
        }] });

/** @deprecated since v3.0.0, will be removed with next major release. Use native CSS Flex instead. */
class PFlex extends BaseComponent {
    alignContent;
    alignItems;
    direction;
    inline;
    justifyContent;
    wrap;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFlex, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PFlex, isStandalone: false, selector: "p-flex,[p-flex]", inputs: { alignContent: "alignContent", alignItems: "alignItems", direction: "direction", inline: "inline", justifyContent: "justifyContent", wrap: "wrap" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFlex, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-flex,[p-flex]',
                    template: '<ng-content />',
                    inputs: ['alignContent', 'alignItems', 'direction', 'inline', 'justifyContent', 'wrap'],
                    standalone: false
                }]
        }] });

/** @deprecated since v3.0.0, will be removed with next major release. Use native CSS Flex instead. */
class PFlexItem extends BaseComponent {
    alignSelf;
    flex;
    grow;
    offset;
    shrink;
    width;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFlexItem, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PFlexItem, isStandalone: false, selector: "p-flex-item,[p-flex-item]", inputs: { alignSelf: "alignSelf", flex: "flex", grow: "grow", offset: "offset", shrink: "shrink", width: "width" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFlexItem, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-flex-item,[p-flex-item]',
                    template: '<ng-content />',
                    inputs: ['alignSelf', 'flex', 'grow', 'offset', 'shrink', 'width'],
                    standalone: false
                }]
        }] });

class PFlyout extends BaseComponentWithTheme {
    aria;
    disableBackdropClick;
    footerBehavior;
    open;
    position;
    theme;
    dismiss = new EventEmitter();
    motionHiddenEnd = new EventEmitter();
    motionVisibleEnd = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFlyout, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PFlyout, isStandalone: false, selector: "p-flyout,[p-flyout]", inputs: { aria: "aria", disableBackdropClick: "disableBackdropClick", footerBehavior: "footerBehavior", open: "open", position: "position", theme: "theme" }, outputs: { dismiss: "dismiss", motionHiddenEnd: "motionHiddenEnd", motionVisibleEnd: "motionVisibleEnd" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFlyout, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-flyout,[p-flyout]',
                    template: '<ng-content />',
                    inputs: ['aria', 'disableBackdropClick', 'footerBehavior', 'open', 'position', 'theme'],
                    outputs: ['dismiss', 'motionHiddenEnd', 'motionVisibleEnd'],
                    standalone: false
                }]
        }] });

class PFlyoutMultilevel extends BaseComponentWithTheme {
    activeIdentifier;
    aria;
    open;
    theme;
    dismiss = new EventEmitter();
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFlyoutMultilevel, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PFlyoutMultilevel, isStandalone: false, selector: "p-flyout-multilevel,[p-flyout-multilevel]", inputs: { activeIdentifier: "activeIdentifier", aria: "aria", open: "open", theme: "theme" }, outputs: { dismiss: "dismiss", update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFlyoutMultilevel, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-flyout-multilevel,[p-flyout-multilevel]',
                    template: '<ng-content />',
                    inputs: ['activeIdentifier', 'aria', 'open', 'theme'],
                    outputs: ['dismiss', 'update'],
                    standalone: false
                }]
        }] });

class PFlyoutMultilevelItem extends BaseComponent {
    cascade;
    identifier;
    label;
    primary;
    secondary;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFlyoutMultilevelItem, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PFlyoutMultilevelItem, isStandalone: false, selector: "p-flyout-multilevel-item,[p-flyout-multilevel-item]", inputs: { cascade: "cascade", identifier: "identifier", label: "label", primary: "primary", secondary: "secondary" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PFlyoutMultilevelItem, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-flyout-multilevel-item,[p-flyout-multilevel-item]',
                    template: '<ng-content />',
                    inputs: ['cascade', 'identifier', 'label', 'primary', 'secondary'],
                    standalone: false
                }]
        }] });

/** @deprecated since v3.0.0, will be removed with next major release. Use native CSS Grid instead. */
class PGrid extends BaseComponent {
    direction;
    /** @deprecated */
    gutter;
    wrap;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PGrid, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PGrid, isStandalone: false, selector: "p-grid,[p-grid]", inputs: { direction: "direction", gutter: "gutter", wrap: "wrap" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PGrid, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-grid,[p-grid]',
                    template: '<ng-content />',
                    inputs: ['direction', 'gutter', 'wrap'],
                    standalone: false
                }]
        }] });

/** @deprecated since v3.0.0, will be removed with next major release. Use native CSS Grid instead. */
class PGridItem extends BaseComponent {
    offset;
    size;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PGridItem, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PGridItem, isStandalone: false, selector: "p-grid-item,[p-grid-item]", inputs: { offset: "offset", size: "size" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PGridItem, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-grid-item,[p-grid-item]',
                    template: '<ng-content />',
                    inputs: ['offset', 'size'],
                    standalone: false
                }]
        }] });

class PHeading extends BaseComponentWithTheme {
    align;
    color;
    ellipsis;
    size;
    tag;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PHeading, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PHeading, isStandalone: false, selector: "p-heading,[p-heading]", inputs: { align: "align", color: "color", ellipsis: "ellipsis", size: "size", tag: "tag", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PHeading, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-heading,[p-heading]',
                    template: '<ng-content />',
                    inputs: ['align', 'color', 'ellipsis', 'size', 'tag', 'theme'],
                    standalone: false
                }]
        }] });

/** @deprecated since v3.0.0, will be removed with next major release. Please use "p-heading" instead. */
class PHeadline extends BaseComponentWithTheme {
    align;
    color;
    ellipsis;
    tag;
    theme;
    variant;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PHeadline, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PHeadline, isStandalone: false, selector: "p-headline,[p-headline]", inputs: { align: "align", color: "color", ellipsis: "ellipsis", tag: "tag", theme: "theme", variant: "variant" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PHeadline, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-headline,[p-headline]',
                    template: '<ng-content />',
                    inputs: ['align', 'color', 'ellipsis', 'tag', 'theme', 'variant'],
                    standalone: false
                }]
        }] });

class PIcon extends BaseComponentWithTheme {
    aria;
    color;
    /** @deprecated */
    lazy;
    name;
    size;
    source;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PIcon, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PIcon, isStandalone: false, selector: "p-icon,[p-icon]", inputs: { aria: "aria", color: "color", lazy: "lazy", name: "name", size: "size", source: "source", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PIcon, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-icon,[p-icon]',
                    template: '<ng-content />',
                    inputs: ['aria', 'color', 'lazy', 'name', 'size', 'source', 'theme'],
                    standalone: false
                }]
        }] });

class PInlineNotification extends BaseComponentWithTheme {
    actionIcon;
    actionLabel;
    actionLoading;
    description;
    dismissButton;
    heading;
    headingTag;
    /** @deprecated */
    persistent;
    state;
    theme;
    action = new EventEmitter();
    dismiss = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PInlineNotification, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PInlineNotification, isStandalone: false, selector: "p-inline-notification,[p-inline-notification]", inputs: { actionIcon: "actionIcon", actionLabel: "actionLabel", actionLoading: "actionLoading", description: "description", dismissButton: "dismissButton", heading: "heading", headingTag: "headingTag", persistent: "persistent", state: "state", theme: "theme" }, outputs: { action: "action", dismiss: "dismiss" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PInlineNotification, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-inline-notification,[p-inline-notification]',
                    template: '<ng-content />',
                    inputs: ['actionIcon', 'actionLabel', 'actionLoading', 'description', 'dismissButton', 'heading', 'headingTag', 'persistent', 'state', 'theme'],
                    outputs: ['action', 'dismiss'],
                    standalone: false
                }]
        }] });

class PLink extends BaseComponentWithTheme {
    aria;
    compact;
    download;
    hideLabel;
    href;
    icon;
    iconSource;
    rel;
    target;
    theme;
    variant;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLink, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PLink, isStandalone: false, selector: "p-link,[p-link]", inputs: { aria: "aria", compact: "compact", download: "download", hideLabel: "hideLabel", href: "href", icon: "icon", iconSource: "iconSource", rel: "rel", target: "target", theme: "theme", variant: "variant" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLink, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-link,[p-link]',
                    template: '<ng-content />',
                    inputs: ['aria', 'compact', 'download', 'hideLabel', 'href', 'icon', 'iconSource', 'rel', 'target', 'theme', 'variant'],
                    standalone: false
                }]
        }] });

class PLinkPure extends BaseComponentWithTheme {
    active;
    alignLabel;
    aria;
    download;
    hideLabel;
    href;
    icon;
    iconSource;
    rel;
    size;
    stretch;
    target;
    theme;
    underline;
    /** @deprecated */
    weight;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLinkPure, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PLinkPure, isStandalone: false, selector: "p-link-pure,[p-link-pure]", inputs: { active: "active", alignLabel: "alignLabel", aria: "aria", download: "download", hideLabel: "hideLabel", href: "href", icon: "icon", iconSource: "iconSource", rel: "rel", size: "size", stretch: "stretch", target: "target", theme: "theme", underline: "underline", weight: "weight" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLinkPure, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-link-pure,[p-link-pure]',
                    template: '<ng-content />',
                    inputs: ['active', 'alignLabel', 'aria', 'download', 'hideLabel', 'href', 'icon', 'iconSource', 'rel', 'size', 'stretch', 'target', 'theme', 'underline', 'weight'],
                    standalone: false
                }]
        }] });

/** @deprecated since v3.0.0, will be removed with next major release. Use `p-link` with corresponding social icon instead. */
class PLinkSocial extends BaseComponentWithTheme {
    compact;
    hideLabel;
    href;
    icon;
    iconSource;
    rel;
    target;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLinkSocial, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PLinkSocial, isStandalone: false, selector: "p-link-social,[p-link-social]", inputs: { compact: "compact", hideLabel: "hideLabel", href: "href", icon: "icon", iconSource: "iconSource", rel: "rel", target: "target", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLinkSocial, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-link-social,[p-link-social]',
                    template: '<ng-content />',
                    inputs: ['compact', 'hideLabel', 'href', 'icon', 'iconSource', 'rel', 'target', 'theme'],
                    standalone: false
                }]
        }] });

class PLinkTile extends BaseComponent {
    align;
    aria;
    aspectRatio;
    background;
    compact;
    description;
    download;
    gradient;
    href;
    label;
    rel;
    size;
    target;
    weight;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLinkTile, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PLinkTile, isStandalone: false, selector: "p-link-tile,[p-link-tile]", inputs: { align: "align", aria: "aria", aspectRatio: "aspectRatio", background: "background", compact: "compact", description: "description", download: "download", gradient: "gradient", href: "href", label: "label", rel: "rel", size: "size", target: "target", weight: "weight" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLinkTile, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-link-tile,[p-link-tile]',
                    template: '<ng-content />',
                    inputs: ['align', 'aria', 'aspectRatio', 'background', 'compact', 'description', 'download', 'gradient', 'href', 'label', 'rel', 'size', 'target', 'weight'],
                    standalone: false
                }]
        }] });

class PLinkTileModelSignature extends BaseComponent {
    aspectRatio;
    description;
    heading;
    headingTag;
    linkDirection;
    model;
    weight;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLinkTileModelSignature, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PLinkTileModelSignature, isStandalone: false, selector: "p-link-tile-model-signature,[p-link-tile-model-signature]", inputs: { aspectRatio: "aspectRatio", description: "description", heading: "heading", headingTag: "headingTag", linkDirection: "linkDirection", model: "model", weight: "weight" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLinkTileModelSignature, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-link-tile-model-signature,[p-link-tile-model-signature]',
                    template: '<ng-content />',
                    inputs: ['aspectRatio', 'description', 'heading', 'headingTag', 'linkDirection', 'model', 'weight'],
                    standalone: false
                }]
        }] });

class PLinkTileProduct extends BaseComponentWithTheme {
    aspectRatio;
    description;
    heading;
    href;
    likeButton;
    liked;
    price;
    priceOriginal;
    rel;
    target;
    theme;
    like = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLinkTileProduct, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PLinkTileProduct, isStandalone: false, selector: "p-link-tile-product,[p-link-tile-product]", inputs: { aspectRatio: "aspectRatio", description: "description", heading: "heading", href: "href", likeButton: "likeButton", liked: "liked", price: "price", priceOriginal: "priceOriginal", rel: "rel", target: "target", theme: "theme" }, outputs: { like: "like" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PLinkTileProduct, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-link-tile-product,[p-link-tile-product]',
                    template: '<ng-content />',
                    inputs: ['aspectRatio', 'description', 'heading', 'href', 'likeButton', 'liked', 'price', 'priceOriginal', 'rel', 'target', 'theme'],
                    outputs: ['like'],
                    standalone: false
                }]
        }] });

/** @deprecated since v3.0.0, will be removed with next major release. Please use "p-wordmark" instead. */
class PMarque extends BaseComponent {
    aria;
    href;
    size;
    target;
    trademark;
    variant;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PMarque, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PMarque, isStandalone: false, selector: "p-marque,[p-marque]", inputs: { aria: "aria", href: "href", size: "size", target: "target", trademark: "trademark", variant: "variant" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PMarque, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-marque,[p-marque]',
                    template: '<ng-content />',
                    inputs: ['aria', 'href', 'size', 'target', 'trademark', 'variant'],
                    standalone: false
                }]
        }] });

class PModal extends BaseComponentWithTheme {
    aria;
    backdrop;
    disableBackdropClick;
    /** @deprecated */
    disableCloseButton;
    dismissButton;
    fullscreen;
    /** @deprecated */
    heading;
    open;
    theme;
    /** @deprecated */
    close = new EventEmitter();
    dismiss = new EventEmitter();
    motionHiddenEnd = new EventEmitter();
    motionVisibleEnd = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PModal, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PModal, isStandalone: false, selector: "p-modal,[p-modal]", inputs: { aria: "aria", backdrop: "backdrop", disableBackdropClick: "disableBackdropClick", disableCloseButton: "disableCloseButton", dismissButton: "dismissButton", fullscreen: "fullscreen", heading: "heading", open: "open", theme: "theme" }, outputs: { close: "close", dismiss: "dismiss", motionHiddenEnd: "motionHiddenEnd", motionVisibleEnd: "motionVisibleEnd" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PModal, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-modal,[p-modal]',
                    template: '<ng-content />',
                    inputs: ['aria', 'backdrop', 'disableBackdropClick', 'disableCloseButton', 'dismissButton', 'fullscreen', 'heading', 'open', 'theme'],
                    outputs: ['close', 'dismiss', 'motionHiddenEnd', 'motionVisibleEnd'],
                    standalone: false
                }]
        }] });

class PModelSignature extends BaseComponentWithTheme {
    color;
    fetchPriority;
    lazy;
    model;
    safeZone;
    size;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PModelSignature, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PModelSignature, isStandalone: false, selector: "p-model-signature,[p-model-signature]", inputs: { color: "color", fetchPriority: "fetchPriority", lazy: "lazy", model: "model", safeZone: "safeZone", size: "size", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PModelSignature, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-model-signature,[p-model-signature]',
                    template: '<ng-content />',
                    inputs: ['color', 'fetchPriority', 'lazy', 'model', 'safeZone', 'size', 'theme'],
                    standalone: false
                }]
        }] });

class PMultiSelect extends BaseComponentWithTheme {
    description;
    disabled;
    dropdownDirection;
    form;
    hideLabel;
    label;
    message;
    name;
    required;
    state;
    theme;
    value;
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PMultiSelect, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PMultiSelect, isStandalone: false, selector: "p-multi-select,[p-multi-select]", inputs: { description: "description", disabled: "disabled", dropdownDirection: "dropdownDirection", form: "form", hideLabel: "hideLabel", label: "label", message: "message", name: "name", required: "required", state: "state", theme: "theme", value: "value" }, outputs: { update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PMultiSelect, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-multi-select,[p-multi-select]',
                    template: '<ng-content />',
                    inputs: ['description', 'disabled', 'dropdownDirection', 'form', 'hideLabel', 'label', 'message', 'name', 'required', 'state', 'theme', 'value'],
                    outputs: ['update'],
                    standalone: false
                }]
        }] });

class PMultiSelectOption extends BaseComponent {
    disabled;
    value;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PMultiSelectOption, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PMultiSelectOption, isStandalone: false, selector: "p-multi-select-option,[p-multi-select-option]", inputs: { disabled: "disabled", value: "value" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PMultiSelectOption, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-multi-select-option,[p-multi-select-option]',
                    template: '<ng-content />',
                    inputs: ['disabled', 'value'],
                    standalone: false
                }]
        }] });

class POptgroup extends BaseComponent {
    disabled;
    label;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: POptgroup, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: POptgroup, isStandalone: false, selector: "p-optgroup,[p-optgroup]", inputs: { disabled: "disabled", label: "label" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: POptgroup, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-optgroup,[p-optgroup]',
                    template: '<ng-content />',
                    inputs: ['disabled', 'label'],
                    standalone: false
                }]
        }] });

class PPagination extends BaseComponentWithTheme {
    activePage;
    /** @deprecated */
    allyLabel;
    /** @deprecated */
    allyLabelNext;
    /** @deprecated */
    allyLabelPage;
    /** @deprecated */
    allyLabelPrev;
    intl;
    itemsPerPage;
    /** @deprecated */
    maxNumberOfPageLinks;
    showLastPage;
    theme;
    totalItemsCount;
    /** @deprecated */
    pageChange = new EventEmitter();
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PPagination, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PPagination, isStandalone: false, selector: "p-pagination,[p-pagination]", inputs: { activePage: "activePage", allyLabel: "allyLabel", allyLabelNext: "allyLabelNext", allyLabelPage: "allyLabelPage", allyLabelPrev: "allyLabelPrev", intl: "intl", itemsPerPage: "itemsPerPage", maxNumberOfPageLinks: "maxNumberOfPageLinks", showLastPage: "showLastPage", theme: "theme", totalItemsCount: "totalItemsCount" }, outputs: { pageChange: "pageChange", update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PPagination, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-pagination,[p-pagination]',
                    template: '<ng-content />',
                    inputs: ['activePage', 'allyLabel', 'allyLabelNext', 'allyLabelPage', 'allyLabelPrev', 'intl', 'itemsPerPage', 'maxNumberOfPageLinks', 'showLastPage', 'theme', 'totalItemsCount'],
                    outputs: ['pageChange', 'update'],
                    standalone: false
                }]
        }] });

class PPinCode extends BaseComponentWithTheme {
    description;
    disabled;
    form;
    hideLabel;
    label;
    length;
    loading;
    message;
    name;
    required;
    state;
    theme;
    type;
    value;
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PPinCode, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PPinCode, isStandalone: false, selector: "p-pin-code,[p-pin-code]", inputs: { description: "description", disabled: "disabled", form: "form", hideLabel: "hideLabel", label: "label", length: "length", loading: "loading", message: "message", name: "name", required: "required", state: "state", theme: "theme", type: "type", value: "value" }, outputs: { update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PPinCode, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-pin-code,[p-pin-code]',
                    template: '<ng-content />',
                    inputs: ['description', 'disabled', 'form', 'hideLabel', 'label', 'length', 'loading', 'message', 'name', 'required', 'state', 'theme', 'type', 'value'],
                    outputs: ['update'],
                    standalone: false
                }]
        }] });

class PPopover extends BaseComponentWithTheme {
    aria;
    description;
    direction;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PPopover, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PPopover, isStandalone: false, selector: "p-popover,[p-popover]", inputs: { aria: "aria", description: "description", direction: "direction", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PPopover, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-popover,[p-popover]',
                    template: '<ng-content />',
                    inputs: ['aria', 'description', 'direction', 'theme'],
                    standalone: false
                }]
        }] });

class PRadioButtonWrapper extends BaseComponentWithTheme {
    hideLabel;
    label;
    loading;
    message;
    state;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PRadioButtonWrapper, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PRadioButtonWrapper, isStandalone: false, selector: "p-radio-button-wrapper,[p-radio-button-wrapper]", inputs: { hideLabel: "hideLabel", label: "label", loading: "loading", message: "message", state: "state", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PRadioButtonWrapper, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-radio-button-wrapper,[p-radio-button-wrapper]',
                    template: '<ng-content />',
                    inputs: ['hideLabel', 'label', 'loading', 'message', 'state', 'theme'],
                    standalone: false
                }]
        }] });

class PScroller extends BaseComponentWithTheme {
    alignScrollIndicator;
    aria;
    gradientColor;
    /** @deprecated */
    gradientColorScheme;
    /** @deprecated */
    scrollIndicatorPosition;
    scrollToPosition;
    scrollbar;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PScroller, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PScroller, isStandalone: false, selector: "p-scroller,[p-scroller]", inputs: { alignScrollIndicator: "alignScrollIndicator", aria: "aria", gradientColor: "gradientColor", gradientColorScheme: "gradientColorScheme", scrollIndicatorPosition: "scrollIndicatorPosition", scrollToPosition: "scrollToPosition", scrollbar: "scrollbar", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PScroller, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-scroller,[p-scroller]',
                    template: '<ng-content />',
                    inputs: ['alignScrollIndicator', 'aria', 'gradientColor', 'gradientColorScheme', 'scrollIndicatorPosition', 'scrollToPosition', 'scrollbar', 'theme'],
                    standalone: false
                }]
        }] });

class PSegmentedControl extends BaseComponentWithTheme {
    /** @deprecated */
    backgroundColor;
    columns;
    disabled;
    form;
    name;
    theme;
    value;
    /** @deprecated */
    segmentedControlChange = new EventEmitter();
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSegmentedControl, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PSegmentedControl, isStandalone: false, selector: "p-segmented-control,[p-segmented-control]", inputs: { backgroundColor: "backgroundColor", columns: "columns", disabled: "disabled", form: "form", name: "name", theme: "theme", value: "value" }, outputs: { segmentedControlChange: "segmentedControlChange", update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSegmentedControl, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-segmented-control,[p-segmented-control]',
                    template: '<ng-content />',
                    inputs: ['backgroundColor', 'columns', 'disabled', 'form', 'name', 'theme', 'value'],
                    outputs: ['segmentedControlChange', 'update'],
                    standalone: false
                }]
        }] });

class PSegmentedControlItem extends BaseComponent {
    aria;
    disabled;
    icon;
    iconSource;
    label;
    value;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSegmentedControlItem, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PSegmentedControlItem, isStandalone: false, selector: "p-segmented-control-item,[p-segmented-control-item]", inputs: { aria: "aria", disabled: "disabled", icon: "icon", iconSource: "iconSource", label: "label", value: "value" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSegmentedControlItem, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-segmented-control-item,[p-segmented-control-item]',
                    template: '<ng-content />',
                    inputs: ['aria', 'disabled', 'icon', 'iconSource', 'label', 'value'],
                    standalone: false
                }]
        }] });

class PSelect extends BaseComponentWithTheme {
    compact;
    description;
    disabled;
    dropdownDirection;
    form;
    hideLabel;
    label;
    message;
    name;
    required;
    state;
    theme;
    value;
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSelect, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PSelect, isStandalone: false, selector: "p-select,[p-select]", inputs: { compact: "compact", description: "description", disabled: "disabled", dropdownDirection: "dropdownDirection", form: "form", hideLabel: "hideLabel", label: "label", message: "message", name: "name", required: "required", state: "state", theme: "theme", value: "value" }, outputs: { update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSelect, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-select,[p-select]',
                    template: '<ng-content />',
                    inputs: ['compact', 'description', 'disabled', 'dropdownDirection', 'form', 'hideLabel', 'label', 'message', 'name', 'required', 'state', 'theme', 'value'],
                    outputs: ['update'],
                    standalone: false
                }]
        }] });

class PSelectOption extends BaseComponent {
    disabled;
    value;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSelectOption, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PSelectOption, isStandalone: false, selector: "p-select-option,[p-select-option]", inputs: { disabled: "disabled", value: "value" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSelectOption, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-select-option,[p-select-option]',
                    template: '<ng-content />',
                    inputs: ['disabled', 'value'],
                    standalone: false
                }]
        }] });

class PSelectWrapper extends BaseComponentWithTheme {
    description;
    dropdownDirection;
    filter;
    hideLabel;
    label;
    message;
    native;
    state;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSelectWrapper, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PSelectWrapper, isStandalone: false, selector: "p-select-wrapper,[p-select-wrapper]", inputs: { description: "description", dropdownDirection: "dropdownDirection", filter: "filter", hideLabel: "hideLabel", label: "label", message: "message", native: "native", state: "state", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSelectWrapper, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-select-wrapper,[p-select-wrapper]',
                    template: '<ng-content />',
                    inputs: ['description', 'dropdownDirection', 'filter', 'hideLabel', 'label', 'message', 'native', 'state', 'theme'],
                    standalone: false
                }]
        }] });

class PSheet extends BaseComponentWithTheme {
    aria;
    disableBackdropClick;
    dismissButton;
    open;
    theme;
    dismiss = new EventEmitter();
    motionHiddenEnd = new EventEmitter();
    motionVisibleEnd = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSheet, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PSheet, isStandalone: false, selector: "p-sheet,[p-sheet]", inputs: { aria: "aria", disableBackdropClick: "disableBackdropClick", dismissButton: "dismissButton", open: "open", theme: "theme" }, outputs: { dismiss: "dismiss", motionHiddenEnd: "motionHiddenEnd", motionVisibleEnd: "motionVisibleEnd" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSheet, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-sheet,[p-sheet]',
                    template: '<ng-content />',
                    inputs: ['aria', 'disableBackdropClick', 'dismissButton', 'open', 'theme'],
                    outputs: ['dismiss', 'motionHiddenEnd', 'motionVisibleEnd'],
                    standalone: false
                }]
        }] });

class PSpinner extends BaseComponentWithTheme {
    aria;
    size;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSpinner, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PSpinner, isStandalone: false, selector: "p-spinner,[p-spinner]", inputs: { aria: "aria", size: "size", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSpinner, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-spinner,[p-spinner]',
                    template: '<ng-content />',
                    inputs: ['aria', 'size', 'theme'],
                    standalone: false
                }]
        }] });

class PStepperHorizontal extends BaseComponentWithTheme {
    size;
    theme;
    /** @deprecated */
    stepChange = new EventEmitter();
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PStepperHorizontal, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PStepperHorizontal, isStandalone: false, selector: "p-stepper-horizontal,[p-stepper-horizontal]", inputs: { size: "size", theme: "theme" }, outputs: { stepChange: "stepChange", update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PStepperHorizontal, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-stepper-horizontal,[p-stepper-horizontal]',
                    template: '<ng-content />',
                    inputs: ['size', 'theme'],
                    outputs: ['stepChange', 'update'],
                    standalone: false
                }]
        }] });

class PStepperHorizontalItem extends BaseComponent {
    disabled;
    state;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PStepperHorizontalItem, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PStepperHorizontalItem, isStandalone: false, selector: "p-stepper-horizontal-item,[p-stepper-horizontal-item]", inputs: { disabled: "disabled", state: "state" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PStepperHorizontalItem, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-stepper-horizontal-item,[p-stepper-horizontal-item]',
                    template: '<ng-content />',
                    inputs: ['disabled', 'state'],
                    standalone: false
                }]
        }] });

class PSwitch extends BaseComponentWithTheme {
    alignLabel;
    checked;
    compact;
    disabled;
    hideLabel;
    loading;
    stretch;
    theme;
    /** @deprecated */
    switchChange = new EventEmitter();
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSwitch, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PSwitch, isStandalone: false, selector: "p-switch,[p-switch]", inputs: { alignLabel: "alignLabel", checked: "checked", compact: "compact", disabled: "disabled", hideLabel: "hideLabel", loading: "loading", stretch: "stretch", theme: "theme" }, outputs: { switchChange: "switchChange", update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PSwitch, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-switch,[p-switch]',
                    template: '<ng-content />',
                    inputs: ['alignLabel', 'checked', 'compact', 'disabled', 'hideLabel', 'loading', 'stretch', 'theme'],
                    outputs: ['switchChange', 'update'],
                    standalone: false
                }]
        }] });

class PTable extends BaseComponentWithTheme {
    caption;
    compact;
    layout;
    theme;
    /** @deprecated */
    sortingChange = new EventEmitter();
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTable, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTable, isStandalone: false, selector: "p-table,[p-table]", inputs: { caption: "caption", compact: "compact", layout: "layout", theme: "theme" }, outputs: { sortingChange: "sortingChange", update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTable, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-table,[p-table]',
                    template: '<ng-content />',
                    inputs: ['caption', 'compact', 'layout', 'theme'],
                    outputs: ['sortingChange', 'update'],
                    standalone: false
                }]
        }] });

class PTableBody extends BaseComponent {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableBody, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTableBody, isStandalone: false, selector: "p-table-body,[p-table-body]", usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableBody, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-table-body,[p-table-body]',
                    template: '<ng-content />',
                    standalone: false
                }]
        }] });

class PTableCell extends BaseComponent {
    multiline;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableCell, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTableCell, isStandalone: false, selector: "p-table-cell,[p-table-cell]", inputs: { multiline: "multiline" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableCell, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-table-cell,[p-table-cell]',
                    template: '<ng-content />',
                    inputs: ['multiline'],
                    standalone: false
                }]
        }] });

class PTableHead extends BaseComponent {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableHead, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTableHead, isStandalone: false, selector: "p-table-head,[p-table-head]", usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableHead, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-table-head,[p-table-head]',
                    template: '<ng-content />',
                    standalone: false
                }]
        }] });

class PTableHeadCell extends BaseComponent {
    hideLabel;
    multiline;
    sort;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableHeadCell, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTableHeadCell, isStandalone: false, selector: "p-table-head-cell,[p-table-head-cell]", inputs: { hideLabel: "hideLabel", multiline: "multiline", sort: "sort" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableHeadCell, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-table-head-cell,[p-table-head-cell]',
                    template: '<ng-content />',
                    inputs: ['hideLabel', 'multiline', 'sort'],
                    standalone: false
                }]
        }] });

class PTableHeadRow extends BaseComponent {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableHeadRow, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTableHeadRow, isStandalone: false, selector: "p-table-head-row,[p-table-head-row]", usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableHeadRow, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-table-head-row,[p-table-head-row]',
                    template: '<ng-content />',
                    standalone: false
                }]
        }] });

class PTableRow extends BaseComponent {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableRow, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTableRow, isStandalone: false, selector: "p-table-row,[p-table-row]", usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTableRow, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-table-row,[p-table-row]',
                    template: '<ng-content />',
                    standalone: false
                }]
        }] });

class PTabs extends BaseComponentWithTheme {
    activeTabIndex;
    gradientColor;
    /** @deprecated */
    gradientColorScheme;
    size;
    theme;
    weight;
    /** @deprecated */
    tabChange = new EventEmitter();
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTabs, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTabs, isStandalone: false, selector: "p-tabs,[p-tabs]", inputs: { activeTabIndex: "activeTabIndex", gradientColor: "gradientColor", gradientColorScheme: "gradientColorScheme", size: "size", theme: "theme", weight: "weight" }, outputs: { tabChange: "tabChange", update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTabs, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-tabs,[p-tabs]',
                    template: '<ng-content />',
                    inputs: ['activeTabIndex', 'gradientColor', 'gradientColorScheme', 'size', 'theme', 'weight'],
                    outputs: ['tabChange', 'update'],
                    standalone: false
                }]
        }] });

class PTabsBar extends BaseComponentWithTheme {
    activeTabIndex;
    gradientColor;
    /** @deprecated */
    gradientColorScheme;
    size;
    theme;
    weight;
    /** @deprecated */
    tabChange = new EventEmitter();
    update = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTabsBar, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTabsBar, isStandalone: false, selector: "p-tabs-bar,[p-tabs-bar]", inputs: { activeTabIndex: "activeTabIndex", gradientColor: "gradientColor", gradientColorScheme: "gradientColorScheme", size: "size", theme: "theme", weight: "weight" }, outputs: { tabChange: "tabChange", update: "update" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTabsBar, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-tabs-bar,[p-tabs-bar]',
                    template: '<ng-content />',
                    inputs: ['activeTabIndex', 'gradientColor', 'gradientColorScheme', 'size', 'theme', 'weight'],
                    outputs: ['tabChange', 'update'],
                    standalone: false
                }]
        }] });

class PTabsItem extends BaseComponent {
    label;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTabsItem, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTabsItem, isStandalone: false, selector: "p-tabs-item,[p-tabs-item]", inputs: { label: "label" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTabsItem, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-tabs-item,[p-tabs-item]',
                    template: '<ng-content />',
                    inputs: ['label'],
                    standalone: false
                }]
        }] });

class PTag extends BaseComponentWithTheme {
    color;
    compact;
    icon;
    iconSource;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTag, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTag, isStandalone: false, selector: "p-tag,[p-tag]", inputs: { color: "color", compact: "compact", icon: "icon", iconSource: "iconSource", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTag, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-tag,[p-tag]',
                    template: '<ng-content />',
                    inputs: ['color', 'compact', 'icon', 'iconSource', 'theme'],
                    standalone: false
                }]
        }] });

class PTagDismissible extends BaseComponentWithTheme {
    aria;
    color;
    label;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTagDismissible, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTagDismissible, isStandalone: false, selector: "p-tag-dismissible,[p-tag-dismissible]", inputs: { aria: "aria", color: "color", label: "label", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTagDismissible, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-tag-dismissible,[p-tag-dismissible]',
                    template: '<ng-content />',
                    inputs: ['aria', 'color', 'label', 'theme'],
                    standalone: false
                }]
        }] });

class PText extends BaseComponentWithTheme {
    align;
    color;
    ellipsis;
    size;
    tag;
    theme;
    weight;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PText, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PText, isStandalone: false, selector: "p-text,[p-text]", inputs: { align: "align", color: "color", ellipsis: "ellipsis", size: "size", tag: "tag", theme: "theme", weight: "weight" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PText, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-text,[p-text]',
                    template: '<ng-content />',
                    inputs: ['align', 'color', 'ellipsis', 'size', 'tag', 'theme', 'weight'],
                    standalone: false
                }]
        }] });

class PTextFieldWrapper extends BaseComponentWithTheme {
    actionIcon;
    actionLoading;
    description;
    hideLabel;
    label;
    message;
    /** @deprecated */
    showCharacterCount;
    showCounter;
    showPasswordToggle;
    state;
    submitButton;
    theme;
    unit;
    unitPosition;
    action = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTextFieldWrapper, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTextFieldWrapper, isStandalone: false, selector: "p-text-field-wrapper,[p-text-field-wrapper]", inputs: { actionIcon: "actionIcon", actionLoading: "actionLoading", description: "description", hideLabel: "hideLabel", label: "label", message: "message", showCharacterCount: "showCharacterCount", showCounter: "showCounter", showPasswordToggle: "showPasswordToggle", state: "state", submitButton: "submitButton", theme: "theme", unit: "unit", unitPosition: "unitPosition" }, outputs: { action: "action" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTextFieldWrapper, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-text-field-wrapper,[p-text-field-wrapper]',
                    template: '<ng-content />',
                    inputs: ['actionIcon', 'actionLoading', 'description', 'hideLabel', 'label', 'message', 'showCharacterCount', 'showCounter', 'showPasswordToggle', 'state', 'submitButton', 'theme', 'unit', 'unitPosition'],
                    outputs: ['action'],
                    standalone: false
                }]
        }] });

class PTextList extends BaseComponentWithTheme {
    /** @deprecated */
    listType;
    /** @deprecated */
    orderType;
    theme;
    type;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTextList, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTextList, isStandalone: false, selector: "p-text-list,[p-text-list]", inputs: { listType: "listType", orderType: "orderType", theme: "theme", type: "type" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTextList, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-text-list,[p-text-list]',
                    template: '<ng-content />',
                    inputs: ['listType', 'orderType', 'theme', 'type'],
                    standalone: false
                }]
        }] });

class PTextListItem extends BaseComponent {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTextListItem, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTextListItem, isStandalone: false, selector: "p-text-list-item,[p-text-list-item]", usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTextListItem, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-text-list-item,[p-text-list-item]',
                    template: '<ng-content />',
                    standalone: false
                }]
        }] });

class PTextarea extends BaseComponentWithTheme {
    autoComplete;
    description;
    disabled;
    form;
    hideLabel;
    label;
    maxLength;
    message;
    minLength;
    name;
    placeholder;
    readOnly;
    required;
    resize;
    rows;
    showCounter;
    spellCheck;
    state;
    theme;
    value;
    wrap;
    blur = new EventEmitter();
    change = new EventEmitter();
    input = new EventEmitter();
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTextarea, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTextarea, isStandalone: false, selector: "p-textarea,[p-textarea]", inputs: { autoComplete: "autoComplete", description: "description", disabled: "disabled", form: "form", hideLabel: "hideLabel", label: "label", maxLength: "maxLength", message: "message", minLength: "minLength", name: "name", placeholder: "placeholder", readOnly: "readOnly", required: "required", resize: "resize", rows: "rows", showCounter: "showCounter", spellCheck: "spellCheck", state: "state", theme: "theme", value: "value", wrap: "wrap" }, outputs: { blur: "blur", change: "change", input: "input" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTextarea, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-textarea,[p-textarea]',
                    template: '<ng-content />',
                    inputs: ['autoComplete', 'description', 'disabled', 'form', 'hideLabel', 'label', 'maxLength', 'message', 'minLength', 'name', 'placeholder', 'readOnly', 'required', 'resize', 'rows', 'showCounter', 'spellCheck', 'state', 'theme', 'value', 'wrap'],
                    outputs: ['blur', 'change', 'input'],
                    standalone: false
                }]
        }] });

class PTextareaWrapper extends BaseComponentWithTheme {
    description;
    hideLabel;
    label;
    message;
    /** @deprecated */
    showCharacterCount;
    showCounter;
    state;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTextareaWrapper, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PTextareaWrapper, isStandalone: false, selector: "p-textarea-wrapper,[p-textarea-wrapper]", inputs: { description: "description", hideLabel: "hideLabel", label: "label", message: "message", showCharacterCount: "showCharacterCount", showCounter: "showCounter", state: "state", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PTextareaWrapper, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-textarea-wrapper,[p-textarea-wrapper]',
                    template: '<ng-content />',
                    inputs: ['description', 'hideLabel', 'label', 'message', 'showCharacterCount', 'showCounter', 'state', 'theme'],
                    standalone: false
                }]
        }] });

class PToast extends BaseComponentWithTheme {
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PToast, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PToast, isStandalone: false, selector: "p-toast,[p-toast]", inputs: { theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PToast, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-toast,[p-toast]',
                    template: '<ng-content />',
                    inputs: ['theme'],
                    standalone: false
                }]
        }] });

class PWordmark extends BaseComponentWithTheme {
    aria;
    href;
    size;
    target;
    theme;
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PWordmark, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.0.5", type: PWordmark, isStandalone: false, selector: "p-wordmark,[p-wordmark]", inputs: { aria: "aria", href: "href", size: "size", target: "target", theme: "theme" }, usesInheritance: true, ngImport: i0, template: '<ng-content />', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PWordmark, decorators: [{
            type: Component,
            args: [{
                    selector: 'p-wordmark,[p-wordmark]',
                    template: '<ng-content />',
                    inputs: ['aria', 'href', 'size', 'target', 'theme'],
                    standalone: false
                }]
        }] });

const DECLARATIONS = [
    PAccordion,
    PBanner,
    PButton,
    PButtonGroup,
    PButtonPure,
    PButtonTile,
    PCanvas,
    PCarousel,
    PCheckbox,
    PCheckboxWrapper,
    PContentWrapper,
    PCrest,
    PDisplay,
    PDivider,
    PFieldset,
    PFieldsetWrapper,
    PFlex,
    PFlexItem,
    PFlyout,
    PFlyoutMultilevel,
    PFlyoutMultilevelItem,
    PGrid,
    PGridItem,
    PHeading,
    PHeadline,
    PIcon,
    PInlineNotification,
    PLink,
    PLinkPure,
    PLinkSocial,
    PLinkTile,
    PLinkTileModelSignature,
    PLinkTileProduct,
    PMarque,
    PModal,
    PModelSignature,
    PMultiSelect,
    PMultiSelectOption,
    POptgroup,
    PPagination,
    PPinCode,
    PPopover,
    PRadioButtonWrapper,
    PScroller,
    PSegmentedControl,
    PSegmentedControlItem,
    PSelect,
    PSelectOption,
    PSelectWrapper,
    PSheet,
    PSpinner,
    PStepperHorizontal,
    PStepperHorizontalItem,
    PSwitch,
    PTable,
    PTableBody,
    PTableCell,
    PTableHead,
    PTableHeadCell,
    PTableHeadRow,
    PTableRow,
    PTabs,
    PTabsBar,
    PTabsItem,
    PTag,
    PTagDismissible,
    PText,
    PTextFieldWrapper,
    PTextList,
    PTextListItem,
    PTextarea,
    PTextareaWrapper,
    PToast,
    PWordmark
];

class DefaultConfig {
    prefix = '';
    theme; // since theme exists on almost every component, it is defined here kind of like a global prop
}
// TODO: unit tests are missing
class PorscheDesignSystemModule {
    theme$ = inject(THEME_TOKEN);
    constructor(configParam) {
        const configs = configParam || [new DefaultConfig()];
        this.theme$.next(configs[0].theme || 'light'); // first config sets the theme
        configs.forEach(load);
    }
    static load(config) {
        return {
            ngModule: PorscheDesignSystemModule,
            providers: [
                {
                    provide: DefaultConfig,
                    multi: true, // to support multiple prefixes in same module
                    useValue: config,
                },
            ],
        };
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PorscheDesignSystemModule, deps: [{ token: DefaultConfig, optional: true }], target: i0.ɵɵFactoryTarget.NgModule });
    static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "19.0.5", ngImport: i0, type: PorscheDesignSystemModule, declarations: [PAccordion, PBanner, PButton, PButtonGroup, PButtonPure, PButtonTile, PCanvas, PCarousel, PCheckbox, PCheckboxWrapper, PContentWrapper, PCrest, PDisplay, PDivider, PFieldset, PFieldsetWrapper, PFlex, PFlexItem, PFlyout, PFlyoutMultilevel, PFlyoutMultilevelItem, PGrid, PGridItem, PHeading, PHeadline, PIcon, PInlineNotification, PLink, PLinkPure, PLinkSocial, PLinkTile, PLinkTileModelSignature, PLinkTileProduct, PMarque, PModal, PModelSignature, PMultiSelect, PMultiSelectOption, POptgroup, PPagination, PPinCode, PPopover, PRadioButtonWrapper, PScroller, PSegmentedControl, PSegmentedControlItem, PSelect, PSelectOption, PSelectWrapper, PSheet, PSpinner, PStepperHorizontal, PStepperHorizontalItem, PSwitch, PTable, PTableBody, PTableCell, PTableHead, PTableHeadCell, PTableHeadRow, PTableRow, PTabs, PTabsBar, PTabsItem, PTag, PTagDismissible, PText, PTextFieldWrapper, PTextList, PTextListItem, PTextarea, PTextareaWrapper, PToast, PWordmark], exports: [PAccordion, PBanner, PButton, PButtonGroup, PButtonPure, PButtonTile, PCanvas, PCarousel, PCheckbox, PCheckboxWrapper, PContentWrapper, PCrest, PDisplay, PDivider, PFieldset, PFieldsetWrapper, PFlex, PFlexItem, PFlyout, PFlyoutMultilevel, PFlyoutMultilevelItem, PGrid, PGridItem, PHeading, PHeadline, PIcon, PInlineNotification, PLink, PLinkPure, PLinkSocial, PLinkTile, PLinkTileModelSignature, PLinkTileProduct, PMarque, PModal, PModelSignature, PMultiSelect, PMultiSelectOption, POptgroup, PPagination, PPinCode, PPopover, PRadioButtonWrapper, PScroller, PSegmentedControl, PSegmentedControlItem, PSelect, PSelectOption, PSelectWrapper, PSheet, PSpinner, PStepperHorizontal, PStepperHorizontalItem, PSwitch, PTable, PTableBody, PTableCell, PTableHead, PTableHeadCell, PTableHeadRow, PTableRow, PTabs, PTabsBar, PTabsItem, PTag, PTagDismissible, PText, PTextFieldWrapper, PTextList, PTextListItem, PTextarea, PTextareaWrapper, PToast, PWordmark] });
    static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PorscheDesignSystemModule, providers: [
            {
                provide: THEME_TOKEN,
                useValue: new BehaviorSubject('light'),
            },
        ] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: PorscheDesignSystemModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: DECLARATIONS,
                    exports: DECLARATIONS,
                    providers: [
                        {
                            provide: THEME_TOKEN,
                            useValue: new BehaviorSubject('light'),
                        },
                    ],
                }]
        }], ctorParameters: () => [{ type: DefaultConfig, decorators: [{
                    type: Optional
                }] }] });

class ToastManager {
    addMessage(message) {
        const toast = document.body.querySelector('p-toast,[p-toast]');
        customElements.whenDefined(toast.tagName.toLowerCase()).then(() => toast.addMessage(message));
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: ToastManager, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
    static ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: ToastManager, providedIn: 'root' });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.0.5", ngImport: i0, type: ToastManager, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { DECLARATIONS, PAccordion, PBanner, PButton, PButtonGroup, PButtonPure, PButtonTile, PCanvas, PCarousel, PCheckbox, PCheckboxWrapper, PContentWrapper, PCrest, PDisplay, PDivider, PFieldset, PFieldsetWrapper, PFlex, PFlexItem, PFlyout, PFlyoutMultilevel, PFlyoutMultilevelItem, PGrid, PGridItem, PHeading, PHeadline, PIcon, PInlineNotification, PLink, PLinkPure, PLinkSocial, PLinkTile, PLinkTileModelSignature, PLinkTileProduct, PMarque, PModal, PModelSignature, PMultiSelect, PMultiSelectOption, POptgroup, PPagination, PPinCode, PPopover, PRadioButtonWrapper, PScroller, PSegmentedControl, PSegmentedControlItem, PSelect, PSelectOption, PSelectWrapper, PSheet, PSpinner, PStepperHorizontal, PStepperHorizontalItem, PSwitch, PTable, PTableBody, PTableCell, PTableHead, PTableHeadCell, PTableHeadRow, PTableRow, PTabs, PTabsBar, PTabsItem, PTag, PTagDismissible, PText, PTextFieldWrapper, PTextList, PTextListItem, PTextarea, PTextareaWrapper, PToast, PWordmark, PorscheDesignSystemModule, THEME_TOKEN, ToastManager };
